test/unit/UpdatesPublisherTest.js

```diff
@@ -0,0 +1,152 @@
+  setupProducerStub.resolves()
+  setupChangesStub.resolves()
+  // spying mocks' methods
+  const producerStopSpy = sandbox.spy(publisher.producer, 'stop')
+  const changesDestroySpy = sandbox.spy(publisher.changes, 'destroy')
+  // launch the publisher event loop
+  publisher.start()
+  t.true(setupProducerStub.notCalled)
+  t.true(setupChangesStub.notCalled)
+  // producer fails
+  publisher.producer.emitError()
+  await Promise.delay(1100)
+  t.true(producerStopSpy.calledOnce)
+  t.true(changesDestroySpy.calledOnce)
+  t.true(setupChangesStub.calledOnce)
+  t.true(setupProducerStub.calledOnce)
+})
+
+test('#start: when changes fails, it restarts, producer does not', async (t) => {
+  // setup all mocks and context
+  const {sandbox, publisher} = t.context
+  publisher.producer = new MockProducer()
+  publisher.changes = new MockChangesStream()
+  // stubbing creation methods
+  const setupProducerStub = sandbox.stub(publisher, 'setupProducer')
+  const setupChangesStub = sandbox.stub(publisher, 'setupChanges')
+  setupProducerStub.resolves()
+  setupChangesStub.resolves()
+  // spying mocks' methods
+  const producerStopSpy = sandbox.spy(publisher.producer, 'stop')
+  const changesDestroySpy = sandbox.spy(publisher.changes, 'destroy')
+  // launch the publisher event loop
+  publisher.start()
+  t.true(setupProducerStub.notCalled)
+  t.true(setupChangesStub.notCalled)
+  // changes fails
+  publisher.changes.emitError()
+  await Promise.delay(1100)
+  t.true(producerStopSpy.notCalled)
+  t.true(setupProducerStub.notCalled)
+  t.true(changesDestroySpy.calledOnce)
+  t.true(setupChangesStub.calledOnce)
+})
+
+test('#start: will successfully detect and publish changes', async (t) => {
+  // setup all mocks and context
+  const {sandbox, publisher} = t.context
+  publisher.producer = new MockProducer()
+  publisher.changes = new MockChangesStream()
+  // stubbing creation methods
+  const setupProducerStub = sandbox.stub(publisher, 'setupProducer')
+  const setupChangesStub = sandbox.stub(publisher, 'setupChanges')
+  setupProducerStub.resolves()
+  setupChangesStub.resolves()
+  // launch the publisher event loop
+  publisher.start()
+  t.true(setupProducerStub.notCalled)
+  t.true(setupChangesStub.notCalled)
+  // changes detected
+  const producerSendMessageSpy = sandbox.spy(publisher.producer, 'sendMessage')
+  publisher.changes.emitData()
+  await Promise.delay(100)
+  t.true(producerSendMessageSpy.calledWith({}))
+})
+
+test('#stop', async (t) => {
+  const {sandbox, publisher} = t.context
+  publisher.producer = new MockProducer()
+  publisher.changes = new MockChangesStream()
+  const producerStopSpy = sandbox.spy(publisher.producer, 'stop')
+  const changesDestroySpy = sandbox.spy(publisher.changes, 'destroy')
+  publisher.start()
+  t.true(producerStopSpy.notCalled)
+  t.true(changesDestroySpy.notCalled)
+  await publisher.stop()
+  t.true(producerStopSpy.calledOnce)
+  t.true(changesDestroySpy.calledOnce)
```

Comment:
For all of your spy assertions, you should be using methods exposed by `sandbox.assert`. The error messages are much better. If `ava` complains about no assertions being made, just add a `t.plan(0)` to the top of the test or alternatively a `t.pass()` at the bottom.